mne/minimum_norm/inverse.py

```diff
@@ -788,26 +818,29 @@ def apply_inverse(evoked, inverse_operator, lambda2=1. / 9.,
     sel = _pick_channels_inverse_operator(evoked.ch_names, inv)
     logger.info('Picked %d channels from the data' % len(sel))
     logger.info('Computing inverse...')
-    K, noise_norm, vertno = _assemble_kernel(inv, label, method, pick_ori)
+    K, noise_norm, vertno, source_nn = _assemble_kernel(inv, label, method,
+                                                        pick_ori)
     sol = np.dot(K, evoked.data[sel])  # apply imaging kernel
 
     is_free_ori = (inverse_operator['source_ori'] ==
-                   FIFF.FIFFV_MNE_FREE_ORI and pick_ori is None)
+                   FIFF.FIFFV_MNE_FREE_ORI and pick_ori != 'normal')
 
-    if is_free_ori:
+    if is_free_ori and pick_ori != 'full':
```

Comment:
To me `fixed=False` means three components should be part of the solution, and `fixed=True` means only one should be (normal component). The `loose` parameter determines a weighting of the three components. The `loose=0` should converge to the solution equivalent to `fixed=True`. If `loose=0` currently returns 1 component per location it should raise an error if `solution='full'`, if it currently returns 3 components per location (2 of which are zero by construction) then it shouldn't raise an error if `solution='full'`.